lib/include/pl/core/ast/ast_node.hpp

```diff
@@ -36,8 +35,7 @@ namespace pl::core::ast {
         [[nodiscard]] bool shouldDocument() const;
 
     private:
-        u32 m_line = 1;
-        u32 m_column = 1;
+        Location m_location = EmptyLocation;
```

Comment:
Instead of having some global variable called EmptyLocation, I'd use something like `Location::Empty` or `Location::Empty()`

Reply:
Would it hurt the perf to use std::optional here?

Reply:
> Instead of having some global variable called EmptyLocation, I'd use something like `Location::Empty` or `Location::Empty()`

Could you give an example on how this would be implemented?